src/main/java/kitchenpos/menu/application/MenuService.java

```diff
@@ -0,0 +1,65 @@
+package kitchenpos.menu.application;
+
+import kitchenpos.menu.domain.Menu;
+import kitchenpos.menu.domain.MenuRepository;
+import kitchenpos.menu.domain.MenuProduct;
+import kitchenpos.menu.domain.MenuProductRepository;
+import kitchenpos.menu.dto.MenuCreateRequest;
+import kitchenpos.menu.dto.MenuMapper;
+import kitchenpos.menu.dto.MenuResponse;
+import kitchenpos.menugroup.MenuGroup;
+import kitchenpos.menugroup.MenuGroupRepository;
+import kitchenpos.product.ProductRepository;
+import org.mapstruct.factory.Mappers;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+public class MenuService {
+    private final MenuRepository menuRepository;
+    private final MenuGroupRepository menuGroupRepository;
+    private final MenuProductRepository menuProductRepository;
+    private final ProductRepository productRepository;
+    private final MenuMapper mapper = Mappers.getMapper(MenuMapper.class);
+
+    public MenuService(
+            final MenuRepository menuRepository,
+            final MenuGroupRepository menuGroupRepository,
+            final MenuProductRepository menuProductRepository,
+            final ProductRepository productRepository
+    ) {
+        this.menuRepository = menuRepository;
+        this.menuGroupRepository = menuGroupRepository;
+        this.menuProductRepository = menuProductRepository;
+        this.productRepository = productRepository;
+    }
+
+    @Transactional
+    public MenuResponse create(final MenuCreateRequest menuCreateRequest) {
+        menuCreateRequest.checkPriceValidation();
+        Optional<MenuGroup> menuGroup = menuGroupRepository.findById(menuCreateRequest.getMenuGroupId());
+        if (!menuGroup.isPresent()) {
+            throw new IllegalArgumentException("invalid menugroup id");
+        }
```

Comment:
아래와 같이 옵셔널을 사용할 수 있겠네요

```suggestion
        MenuGroup menuGroup = menuGroupRepository.findById(menuCreateRequest.getMenuGroupId())
        .orElseThrow(() -> new IllegalArgumentException("invalid menugroup id"));
```